### PR TITLE
Sync thermostat device type with specs

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
What is being fixed?  Examples:
* Thermostat device type was not matching specs

#### Change overview
* Sync thermostat device type with specs.
* Added attributes and command for time synchronisation cluster

#### Testing
How was this tested? (at least one bullet point required)
* Verified  thermostat device type syncs with specs

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -1833,7 +1833,7 @@ limitations under the License.
                 <requireCommand>GetSceneMembership</requireCommand>
                 <requireCommand>GetSceneMembershipResponse</requireCommand>
             </include>
-            <include cluster="Groups" client="false" server="true" clientLocked="true" serverLocked="true">
+            <include cluster="Groups" client="false" server="true" clientLocked="true" serverLocked="false">
```
Comment (dhrishi) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1836]: Groups Server, Conformance -> Awake. What does it mean?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -1810,10 +1810,10 @@ limitations under the License.
                 <requireAttribute>CLIENT_LIST</requireAttribute>
                 <requireAttribute>PARTS_LIST</requireAttribute>
             </include>
-            <include cluster="Binding" client="true" server="false" clientLocked="false" serverLocked="true">
+            <include cluster="Binding" client="false" server="true" clientLocked="true" serverLocked="true">
```
Comment (dhrishi) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1813]: Binding client is enabled due to the mandatory time sync client, right?

Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1813]: Yes

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1836]: Optional but recommend ?

Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Awake is optional.

**Review by woody-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 3]
```diff
@@ -1847,11 +1847,12 @@ limitations under the License.
                 <requireCommand>AddGroupIfIdentifying</requireCommand>
             </include>
             <include cluster="Thermostat" client="false" server="true" clientLocked="true" serverLocked="true"></include>
+	    <include cluster="Time Synchronization" client="true" server="true" clientLocked="true" serverLocked="true"></include>
```
Comment (dhrishi) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1850]: Time Sync server and client optional `[Matter]`

**Review by dhrishi — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 3 above)
Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Done

**Review by dhrishi — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**